pkg/controller/kibana/controller.go

```diff
@@ -60,8 +63,13 @@ var (
 )
 
 func (r *ReconcileKibana) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	es, err := k8shandler.FindElasticserachCR(r.client, request.Namespace)
+	if err != nil {
+		return reconcileResult, fmt.Errorf("skipping kibana reconciliation in %q: %s", request.Namespace, err)
+	}
```

Comment:
seeing as how we do this for every kibana reconcile, wouldn't it make sense to have this logic in that function instead of having it in three different places?

Reply:
This breaks dependency injection. I would rather keep the deps out of the k8shandler packages.

Reply:
is there another place we can put it on the controller level? we repeat this code in 4 places and if we ever need to update it there's a risk we'll miss a place

Reply:
TBH the right approach would be a single kibana controller watch for the three objects proxyconfig, secret and kibana CR.